### PR TITLE
chore: fix website staging deploy

### PR DESCRIPTION
**Pull request by olizilla**
the dep on `web3.storage` fails `npm run build` in the website.

License: (Apache-2.0 AND MIT)
Signed-off-by: Oli Evans <oli@tableflip.io>

### CODE REVIEW

**Comment by olizilla** [comment]
Removed the guard on deploying the staging site so i can stest it from this PR.

**Comment by olizilla** [comment]
oh i give up https://github.com/web3-storage/web3.storage/pull/47/checks?check_run_id=3021495277

File: packages/website/package.json [hunk 1]
```diff
@@ -18,8 +18,7 @@
     "react-dom": "^17.0.2",
     "react-if": "^4.0.1",
     "react-query": "^3.13.9",
-    "tachyons": "^4.12.0",
-    "web3.storage": "^0.0.0"
```
Comment (jonyg80) [inline packages/website/package.json:22]: Maybe 
```
"web3.storage": "^1.0.0"
```
From https://github.com/web3-storage/web3.storage/blob/55162eeb632eebf01452e79b8d95ae5248a0f748/package.json#L3

**Comment by olizilla** [comment]
Locally, running:

```console
$ rm -rf node_modules
$ npm ci
$ npm rum build --workspace packages/website
```
builds the website without error. In CI the next.js build fails, claiming:
```console
Failed to compile.

./pages/files.js:4:29
Type error: Cannot find module 'web3.storage' or its corresponding type declarations.

  2 | import { useQuery, useQueryClient } from 'react-query'
  3 | import filesize from 'filesize'
> 4 | import { Web3Storage } from 'web3.storage'
    |                             ^
  5 | import Button from '../components/button.js'
  6 | import Loading from '../components/loading'
  7 | import { getUploads, getToken, API } from '../lib/api.js'
npm ERR! Lifecycle script `build` failed with error: 
```

**Comment by olizilla** [comment]
AH HA! The `client` project has have it's build step run for the types to be available over in the website build. _oh my_

**Comment by olizilla** [comment]
I'd like to take a moment to thank the makers of `act` https://github.com/nektos/act for letting me debug this locally.

**Comment by olizilla** [comment]
Huzzah! it builds! and pins! now we are on to good old fashioned Cloudflare API token errors.

**Comment by olizilla** [comment]
staging is deployed: https://staging.web3.storage/

https needed fixing by flipping the staging dns record to be proxied by cloudflare, and setting the TLS mode to flexible. not sure why the gateway's ssl cert is not suffcient to allow full to be enabled, but we can move to pointing at our cluster and fix that.

**Comment by olizilla** [comment]
Still need to fix ENV for the website.
> Magic SDK Error: [MISSING_API_KEY]

**Comment by olizilla** [comment]
hmm, now a CORS error after logging in when using the ipfs preview site (still waiting for dns ttl to expire on dnslinked domain)

> Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://api-staging.web3.storage/user/login. (Reason: CORS header 'Access-Control-Allow-Origin' missing).

**Comment by olizilla** [comment]
Ah, the CORS error is actually a 500 response (that lacks CORS headers... maybe we should send them on 500s to make debugging easier)

Screenshot 2021-07-09 at 09 14 44

**Comment by olizilla** [comment]
Updated `MAGIC_SECRET_KEY` for the staging api to use the new staging config

**Comment by olizilla** [comment]
argggg coooommmeee oonnnnnn waaaaat
https://github.com/web3-storage/web3.storage/blob/70ac5b41f543bb24b62beb0de557f2103c212e24/packages/website/lib/constants.js#L4-L17

**Comment by olizilla** [comment]
AH HA! now i need to go create the staging schema in FaunaDB! (TODO: want to set that up from CI too.)

**Comment by olizilla** [comment]
Blocked on FaunaDB import error https://github.com/web3-storage/web3.storage/issues/49

**Comment by alanshaw** [comment]
😱 oh no my bad (probably)

**Comment by olizilla** [comment]
Fix in #50

**Comment by olizilla** [comment]
Holy heck. IT LIVES!!!

Screenshot 2021-07-09 at 11 44 40

**Comment by olizilla** [comment]
Checking to see if I can bring back the IPFS commit status message when pinning to cluster is complete.

**Comment by olizilla** [comment]
Ok, is nice now. We pass the cid as output from the build job that is used in the deploy job to update the dnslink if it's a merge to main.

**Comment by olizilla** [comment]
Am merging this. Eyeball at your leisure.

**Review by alanshaw — APPROVED** [review]

**Pull request merged**